### PR TITLE
Implemented switchToIframe method for InnerBrowser

### PR DESCRIPTION
**Pull request by Naktibalda**
#2398

### CODE REVIEW

**Comment by DavertMik** [comment]
Yes, we need to do something to switch back. The solution of WebDriver module is not the something I really like - if name is empty we are switching to main page. Maybe a new method like `switchToPage` sounds good?

It will work with `sendAjaxRequest` and `switchToIFrame`...

Actually I don't know if someone really uses `sendAjaxRequest` methods to load HTML pages partially. But probably it is used and we can improve it.

**Comment by Naktibalda** [comment]
That's the meaning of Page? Do you mean getOutOfFrames?

How about implementing moveBack() method?
Since Javascript is not supported, there is not much state to be kept.

But if we wanted to keep multiple frames in memory, we would have to keep separate instances of module and connector for each frame (to have separate histories, form fields and whatever state there is).

**Comment by DavertMik** [comment]
> But if we wanted to keep multiple frames in memory, we would have to keep separate instances of module and connector for each frame (to have separate histories, form fields and whatever state there is).

No.We should keep only one state and only for cases where we narrow context: like we do for iframes or ajax requests. That would be reasonable as I think of it

*

File: src/Codeception/Lib/InnerBrowser.php [hunk 1]
```diff
@@ -1300,12 +1295,16 @@ protected function clientRequest($method, $uri, array $parameters = array(), arr
         }
 
         if (!PropertyAccess::readPrivateProperty($this->client, 'followRedirects')) {
-            return $this->client->request($method, $uri, $parameters, $files, $server, $content, $changeHistory);
+            $result = $this->client->request($method, $uri, $parameters, $files, $server, $content, $changeHistory);
+            $this->debugResponse();
+            return $result;
+        } else {
```
Comment (DavertMik) [inline src/Codeception/Lib/InnerBrowser.php]: else is not needed

File: tests/unit/Codeception/Module/TestsForWeb.php [hunk 2]
```diff
@@ -1292,4 +1292,12 @@ public function testFillFieldWithAmpersand()
         $form = data::get('form');
         $this->assertEquals('this & that', $form['name']);
     }
+
+    public function testSwitchToIframe()
+    {
+        $this->module->amOnPage('/iframe');
+        $this->module->switchToIframe('content');
+        $this->module->see('Is that interesting?');
+        $this->module->click('Ссылочка');
```
Comment (sergeyklay) [inline tests/unit/Codeception/Module/TestsForWeb.php]: > Ссылочка

:)

Comment (Naktibalda) [inline tests/unit/Codeception/Module/TestsForWeb.php]: I tried to click "Sign in!", but it failed with exception.

Comment (sergeyklay) [inline tests/unit/Codeception/Module/TestsForWeb.php]:
```sh
$ ~/projects/php/Codeception grep -nir 'Ссылочка'
tests/data/app/view/info.php:28:<a href="/">Ссылочка</a>
tests/unit/Codeception/Module/PhpBrowserTest.php:67:        $this->module->seeLink('Ссылочка');
tests/unit/Codeception/Module/PhpBrowserTest.php:68:        $this->module->click('Ссылочка');
```

Comment (Naktibalda) [inline tests/unit/Codeception/Module/TestsForWeb.php]: I got a very misleading exception: [MalformedLocatorException] XPath or CSS locator is malformed: Sign In! 
I can see why it failed now.

**Comment by DavertMik** [comment]
I think this can be merged. Please do it if you don't find anything else should be added

**Pull request merged**